java/sage/EPG.java

```diff
@@ -1324,38 +1329,38 @@ public int getServiceLevel(long providerID, int stationID)
   {
     synchronized (lineups)
     {
-      java.util.Map lineMap = (java.util.Map) lineups.get(new Long(providerID));
+      java.util.Map<Integer, String[]> lineMap = lineups.get(new Long(providerID));
       if (lineMap == null) return new int[0];
       int[] rv = new int[lineMap.keySet().size()];
-      java.util.Iterator walker = lineMap.keySet().iterator();
+      java.util.Iterator<Integer> walker = lineMap.keySet().iterator();
       int idx = 0;
       while (walker.hasNext())
-        rv[idx++] = ((Integer) walker.next()).intValue();
+        rv[idx++] = walker.next().intValue();
       return rv;
     }
   }
 
-  private static java.util.Map parseChanMapString(String str)
+  private static java.util.Map<Long, Integer> parseChanMapString(String str)
```

Comment:
You changed this from being an Integer, Integer map to being a Long, Integer map. It should be an Integer, Integer map because it's mapping station IDs to service levels and station IDs are Integer values....there's a few spots where this needs to be changed to resolve that.